app/models/version.rb

```diff
@@ -0,0 +1,62 @@
+class Version < ActiveRecord::Base
+  belongs_to :target, polymorphic: true
+
+  validate :numbers_must_increase
+  validate :draft_cannot_be_behind_live
+  validate :live_is_equal_to_draft
```

Comment:
This raises a flag for me because this will only necessarily be true at the point the live version is saved. If this record is loaded from the database in future, a newer draft could have been saved, and then this record won't be valid any more.